gocd/templates/bash/check-sentry-errors.sh

```diff
@@ -1,21 +1,38 @@
 #!/bin/bash
 
+# Check Sentry for errors in the last 5 minutes for a given release (from GO_REVISION_RELAY_REPO)
+#
+# Required environment variables:
+#   DRY_RUN: When dry-run is 'true' this script will not fail if the checks indicate an issue
+#   ERROR_LIMIT: The number of error events to permit before failing
+#   GO_REVISION_RELAY_REPO: Git commit hash (provided by GoCD)
+#   SENTRY_AUTH_TOKEN: Sentry auth token (https://sentry.io/settings/account/api/auth-tokens/) (required by devinfra/scripts/checks/sentry/release_error_events.py)
+#   SENTRY_BASE: Sentry base API URL (e.g. https://sentry.io/api/0)
+#   SENTRY_PROJECTS: A space-separated list of <project_id>:<project_slug>:<service> tuples
+#   SENTRY_SINGLE_TENANT: When single-tenant is 'true' this script will use the sentry-st organization instead of sentry
+#   SKIP_CANARY_CHECKS: Whether to skip checks entirely (true/false)
+
 # shellcheck disable=SC2206
-project_ids=(${SENTRY_PROJECT_IDS})
-# shellcheck disable=SC2206
-project_slugs=(${SENTRY_PROJECTS})
+projects=(${SENTRY_PROJECTS})
 
+for project in "${projects[@]}"; do
+  IFS=':' read -r -a project_info <<<"$project"
+  project_id="${project_info[0]}"
+  project_slug="${project_info[1]}"
+  service="${project_info[2]}"
 
-if [ ${##project_ids[@]} -ne ${##project_slugs[@]} ]; then
-  echo "Error: SENTRY_PROJECT_IDS and SENTRY_PROJECTS must have the same number of elements"
-  exit 1
-fi
+  release_name=$(./relay/scripts/get-sentry-release-name "${GO_REVISION_RELAY_REPO}" "${service}")
+  if [ -z "${release_name}" ]; then
+    echo "Failed to get the release name for ${service} at ${GO_REVISION_RELAY_REPO}"
+    # Since Processing and PoPs can be deployed independently, we shouldn't fail if
+    # we can't find a release as it may not exist yet
+    continue
+  fi
 
-for i in "${!project_ids[@]}"; do
   /devinfra/scripts/checks/sentry/release_error_events.py \
-    --project-id="${project_ids[i]}" \
-    --project-slug="${project_slugs[i]}" \
-    --release="relay@${GO_REVISION_RELAY_REPO}" \
+    --project-id="${project_id}" \
+    --project-slug="${project_slug}" \
```

Comment:
Why does `release_error_events` need the project slug? Isn't the project ID enough to uniquely identify the project?

Reply:
It uses two different api calls each of which require different unique identifiers.